sycl/test-e2e/Regression/default-constructed-local-accessor.cpp

```diff
@@ -0,0 +1,18 @@
+// RUN: %{build} -o %t.out -O0
+// RUN: %{run} %t.out
```

Comment:
`-O0` is necessary; on higher levels of optimization, an error does not occur because of dead argument elimination of the `local_accessor`.

Reply:
This comment should be moved to the test itself